### PR TITLE
dnsdist/rec: Drop remaining capabilities after startup

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Drop any remaining capabilities after startup, for example if we have been started as root without `--uid` or `--gid` (please don't do that) or as an unprivileged user with ambient capabilities like `CAP_NET_BIND_SERVICE`.
This does change the existing behavior for setups not using `--uid` or `--gid`, because `dnsdist` used to keep `CAP_NET_BIND_SERVICE` around in that case, and won't after this PR is merged. The only case I can think of right now where it might matter is someone willing to start the webserver on a privileged port after startup, using the console.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Added the same mechanism for the recursor.

**Comment by pieterlexis** [comment]
> This does change the existing behavior for setups not using --uid or --gid, because dnsdist used to keep CAP_NET_BIND_SERVICE around in that case, and won't after this PR is merged.

This should be documented somewhere indeed.

Another thing is that this capability should be added to the spec files, dockerfiles and debian/rules etc. so it is actually used in our packages (I can PR that if you want)

**Comment by rgacogne** [comment]
> Another thing is that this capability should be added to the spec files, dockerfiles and debian/rules etc. so it is actually used in our packages (I can PR that if you want)

Do you know how dnsdist is currently started in our packages? With `--uid dnsdist` ? Do we create the `dnsdist` user using sysusers where available?

**Comment by pieterlexis** [comment]
> Do you know how dnsdist is currently started in our packages? With --uid dnsdist ? Do we create the dnsdist user using sysusers where available?

1. yes ([debian](https://github.com/PowerDNS/pdns/blob/c2ca8dc0dbf73969759a5812deff0ddb4990d363/builder-support/debian/dnsdist/debian-stretch/rules#L57), [centos](https://github.com/PowerDNS/pdns/blob/c2ca8dc0dbf73969759a5812deff0ddb4990d363/builder-support/specs/dnsdist.spec#L125-L126)
2. no sysusers atm, they are created in the postinstall

> Same question for the rec :) I can check if you don't know, of course :)

Similar answers as above :)

File: pdns/recursordist/Makefile.am [hunk 1]
```diff
@@ -369,6 +371,9 @@ rec_control_SOURCES = \
 	rec_control.cc \
 	unix_utility.cc
 
+rec_control_LDADD = \
```
Comment (pieterlexis) [inline pdns/recursordist/Makefile.am]: I don't see why we'd need libcap support in `rec_control` (apart from "this is `#include`d because reasons")

Comment (rgacogne) [reply pdns/recursordist/Makefile.am]: Because it links `misc.o` in, which now refers to symbols from `libcap`.

Comment (rgacogne) [reply pdns/recursordist/Makefile.am]: Fixed!

**Pull request merged**